code/modules/jobs/job_types/job.dm

```diff
@@ -44,15 +44,17 @@
 	var/outfit = null
 
 //Only override this proc
-/datum/job/proc/after_spawn(mob/living/carbon/human/H)
+//H is usually a human unless an /equip override transformed it
+/datum/job/proc/after_spawn(mob/living/H, mob/M)
+	//do actions on H but send messages to M as the key may not have been transferred_yet

```

Comment:
I don't know if it worked but i like the added explicit link between to be spawned character.